checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectTypeFactory.java

```diff
@@ -550,6 +562,11 @@ public void constrainLambdaToUI(LambdaExpressionTree lambdaExpressionTree) {
         uiLambdas.add(lambdaExpressionTree);
     }
 
+    public void constrainAnonymousClassToUI(TypeElement callerReceiverElt) {
```

Comment:
Maybe also adapt a copy of the `constrainLambdaToUI` javadoc?

Reply:
Done in next commit.

Reply:
@lazaroclapp Sorry for the delay on this. Do you have a local commit with more javadoc and forgot to push?
Thanks for commenting on the other issues and finding the earlier discussions. I would suggest shelving those cleanups for a future PR.
Please add the javadoc for the new functionality and maybe add some comments for the other parts; then merge master and we can merge this.